packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,687 @@
+
+*index.js:*
+```jsx
+...
+import {
+  IntegratedFiltering
+} from '@devexpress/dx-react-grid';
+
+class App extends Component {
+  ...
+  render() {
+    ...
+    return (
+      <Paper>
+        <Grid
+          ...
+        >
+          <ToolbarFilterState defaultColumnName={"CompanyName"} defaultFilterValue={""}/>
+          <IntegratedFiltering />
+          <Table />
+          <TableHeaderRow />
+          <Toolbar />
+          <ToolbarFilter />
+        </Grid>
+      </Paper>
+    );
+  }
+}
+```
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+
+const filterExpressionComputed = ({filterExpression, toolbarFilterValue, toolbarFilterColumnName}) => {
+    const newFilterExpression = { ...filterExpression || {filters: [], operator: "and"} };
+    if (toolbarFilterValue) {
+      newFilterExpression.filters.push({
+        columnName: toolbarFilterColumnName,
+        value: toolbarFilterValue
+      });
+    }
+    return newFilterExpression;
+}
+
+export class ToolbarFilterState extends React.PureComponent {
+  ...
+  render() {
+    ...
+    return (
+      <Plugin
+        name="ToolbarFilterState"
+      >
+        ...
+        <Getter name="filterExpression" computed={filterExpressionComputed} />
+      </Plugin>
+    );
+  }
+}
+```
+
+Now our grid can filter items by the `CompanyName` column.
+
+
+
+## Clearing The Current Filter Value And Displaying The Current Column Caption
+
+Let's complete this task by displaying the current column caption near the select box and adding the `Clear` button for clearing the selected value. For this task we will perform the following steps:
+
+* Declare the `clearToolbarFilterValue` action in the `ToolbarFilterState` plugin and clear the `filterValue` state property in the action.
+* Add the `Button` widget near the select box in the `ToolbarFilter` plugin and use the `clearToolbarFilterValue` action as the `onClick` event handler.
+* Get the `toolbarFilterColumnName` getter value in the `ToolbarFilterState` plugin.
+* Since we want to get a caption of the `CompanyName` column, we will need to find the required column by its name and get the caption. For this task we will get the `columns` getter value that is exported by the [Grid](../reference/grid.md) plugin.
+
+The following code snippets show this approach in action:
+*file toolbar-filter-state.js:*
+```jsx
+...
+
+export class ToolbarFilterState extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    ...
+    this.columnName = this.props.defaultColumnName || null;
+    this.clearValue = this.clearValue.bind(this);
+  }
+
+  clearValue() {
+    this.setState({
+      filterValue: ""
+    });
+  }
+
+  render() {
+    ...
+    return (
+      <Plugin
+        name="ToolbarFilterState"
+      >
+        ...
+        <Action name="clearToolbarFilterValue" action={this.clearValue} />
+      </Plugin>
+    );
+  }
+}
+```
+
+*toolbar-filter.js:*
+```jsx
+...
+const columnTitle = function(columns, toolbarFilterColumnName) {
```

Comment:
I suggest to rename `columnTitle` -> `getColumnTitle`

Reply:
Also, we prefer '() => {}' instead of 'function';